File: march-2026/3-03-blood-moon/blood_moon.py
## Blood Moon 🩸
## Codédex

def blood_moon(time):
  # Write code below 💖

  hrs_string, min_string = time.split(":")
  result = []
  
  hrs = int(hrs_string)
  min = int(min_string)

  # print(hrs)
  # print(min)
  
  # ====== first one ====== 

  hrs = hrs + 2
  min = min + 48

  if min > 60:
    min = min - 60
    hrs = hrs + 1
  
  if hrs > 24:
    hrs = hrs - 24
  
  result.append(f"{hrs:02d}:{min:02d}")

  # ====== second one ====== 

  hrs = hrs + 2
  min = min + 48

  if min > 60:
    min = min - 60
    hrs = hrs + 1

  if hrs > 24:
    hrs = hrs - 24
  
  result.append(f"{hrs:02d}:{min:02d}")

  # ====== third one ====== 

  hrs = hrs + 2
  min = min + 48

  if min > 60:
    min = min - 60
    hrs = hrs + 1
  
  if hrs > 24:
    hrs = hrs - 24
  
  result.append(f"{hrs:02d}:{min:02d}")

  return result

File: march-2026/3-03-blood-moon/bloodMoon.js
// Blood Moon 🩸
// Codédex

function bloodMoon(time) {
  
  const result = []; 
  const [hours, minutes] = time.split(":").map(Number);

  let date = new Date();
  date.setHours(hours, minutes, 0, 0);

  for (let i = 0; i < 3; i++) {
    date.setMinutes(date.getMinutes() + 168); 
    const hour = String(date.getHours()).padStart(2, "0");
    const minute = String(date.getMinutes()).padStart(2, "0");
    result.push(`${hour}:${minute}`);
  }

  return result; 
}

console.log(bloodMoon("22:30"))